src/universal/modules/welcome/containers/Welcome/Welcome.js

```diff
@@ -32,16 +49,27 @@ const mapStateToProps = state => ({
 export default class WelcomeContainer extends Component {
   static propTypes = {
     dispatch: PropTypes.func,
+    invitees: PropTypes.array,
+    inviteesRaw: PropTypes.string,
+    preferredName: PropTypes.string,
+    teamName: PropTypes.string,
     welcome: PropTypes.shape({
-      preferredName: PropTypes.string,
-      teamName: PropTypes.string,
       teamId: PropTypes.string,
       teamMemberId: PropTypes.string
     })
   };
 
+  constructor(props) {
+    super(props);
+    this.onPreferredNameSubmit = this.onPreferredNameSubmit.bind(this);
```

Comment:
correct me if I'm wrong, but since you take full advantage of those fatty little arrows: `onPreferredNameSubmit = data => {`, you don't need to bind nothin